packages/dds/tree/src/util/typeUtils.ts

```diff
@@ -137,3 +137,90 @@ export type RestrictiveStringRecord<T> = {
 } & {
 	readonly [P in symbol]?: never;
 };
+
+/**
+ * Returns `true` if T is a union and false if it is not.
+ * @typeparam T - Type to test if it is a union.
+ * @typeparam T2 - Do not specify: default value used as implementation detail.
+ * @system @alpha
+ */
+export type IsUnion<T, T2 = T> = T extends unknown
+	? [T2] extends [T]
+		? false
+		: true
+	: "error";
+
+/**
+ * Convert a union of types to an intersection of those types. Useful for `TransformEvents`.
+ * @privateRemarks
+ * First an always true extends clause is used (T extends T) to distribute T into to a union of types contravariant over each member of the T union.
+ * Then the constraint on the type parameter in this new context is inferred, giving the intersection.
+ * @system @public
+ */
+export type UnionToIntersection<T> = (T extends T ? (k: T) => unknown : never) extends (
+	k: infer U,
+) => unknown
+	? U
+	: never;
+
+/**
+ * Gets the first item of a union type.
+ *
+ * @typeparam Union - The union to convert.
+ * @typeparam AsOverloadedFunction - Implementation detail: do not specify.
+ * @system @alpha
+ */
+export type PopUnion<
+	Union,
+	AsOverloadedFunction = UnionToIntersection<
+		Union extends unknown ? (f: Union) => void : never
+	>,
+> = AsOverloadedFunction extends (a: infer First) => void ? First : never;
+
+/**
+ * Converts a union type to a tuple type.
+ *
+ * @typeparam T - The union to convert.
+ * @typeparam A - Implementation detail: do not specify.
+ * @typeparam First - Implementation detail: do not specify.
+ *
+ * @privateRemarks
+ * `A` is a tuple to prepend members of the union to.
+ *
+ * https://www.hacklewayne.com/typescript-convert-union-to-tuple-array-yes-but-how and https://catchts.com/union-array both explain the general approach this uses pretty well.
```

Comment:
Nit
```suggestion
 * {@link https://www.hacklewayne.com/typescript-convert-union-to-tuple-array-yes-but-how} and {@link https://catchts.com/union-array} both explain the general approach this uses pretty well.
```

Reply:
Is there a reason to use link tags in privateRemarks? Links are clickable in IntelliSense and inline without them, and they don't end up in the generated docs.

Reply:
In theory so that refactoring tools update the text if the target changes.

Reply:
If you mean the target URL, I'm not sure how a refactoring tools would do that (no one is using a refactoring tool that updates a third party sit's URL scheme and our codebase at the same time), and if you mean the link display text, links in this format can't have that's not just the url so I don't think that applies here either.

Is there so other refactoring tool that I'm missing here?

Reply:
Sorry, I missed that this one was a URL link. My comment applies to symbolic links in private remarks. For URLs, it probably doesn't really matter so long as intellisense handles them well without the tag.